<launch>
  <arg name="camera" default="diff_cam_A"/>
  
  <param name="vision/cam_prefix" value="A" />

  <node pkg="diff-rs-pcl2" type="D-main-a" name="diff_A" output="screen">
    
  <!-- Step 1 - CUBE SEGMENTATION  -->
    <remap from="diff_A/input" to="/$(arg camera)/depth/color/points" />
    <remap from="cube_A/output" to="/$(arg camera)/differential/ladybug/cube" />
    
    <!-- Filter Params for Y axis -->
    <param name="filter_low_y" value="-0.05" />
    <param name="filter_high_y" value="0.06" /> 
    <!-- Filter Params for Z axis -->
    <param name="filter_low_z" value="0" />
    <param name="filter_high_z" value="1" />
    <!-- Filter Params for X axis -->
    <param name="filter_low_x" value="-0.5" />
    <param name="filter_high_x" value="0.5" />


  <!-- Step 2 - DBSCAN Cluster -->
    <remap from="dbscan/input" to="/$(arg camera)/depth/color/points" />
    <!-- DBSCAN Params -->
    <param name="cluster_tolerance" value="0.01" />
    <param name="min_cluster_size" value="200" />
    <param name="max_cluster_size" value="20000" />

  <!-- Step 3 - Marker Array Params -->
    <param name="line_thickness" value="0.005" />

  <!-- Step 4 - Safety Params -->
    <!-- unit: m -->
    <param name="min_safety_dist" value="0.4" />
    <!-- unit: sec -->
    <param name="obstacle_existence_checking_period" value="1" /> 
  </node>
</launch>
